src/main/java/org/opentripplanner/middleware/controllers/api/AbstractUserController.java

```diff
@@ -118,6 +119,15 @@ U preCreateHook(U user, Request req) {
     @Override
     U preUpdateHook(U user, U preExistingUser, Request req) {
         Auth0Users.validateExistingUser(user, preExistingUser, req, this.persistence);
+        if (user instanceof OtpUser) {
+            OtpUser otpUser = (OtpUser) user;
+            OtpUser existingOtpUser = (OtpUser) preExistingUser;
+            if(!otpUser.storeTripHistory && existingOtpUser.storeTripHistory) {
+                // If an OTP user no longer wants their trip history stored, remove all history from MongoDB and all
+                // files previously uploaded to S3.
+                ConnectedDataManager.removeUsersTripHistory(otpUser.id);
+            }
+        }
```

Comment:
I was wondering if a user's data should also be removed if they delete their account, but I suppose this approach is best so that users would specifically note that they want to remove their trip history and then they can delete their account. I'm not even sure if it's possible to delete an account tho, lol.